File: PART3/ch02/ch02_02/service/api/src/main/java/org/delivery/api/common/error/ErrorCode.java
package org.delivery.api.common.error;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.http.HttpStatus;

@AllArgsConstructor
@Getter
public enum ErrorCode implements ErrorCodeIfs {

    OK(200 , 200 , "성공"),
    BAD_REQUEST(HttpStatus.BAD_REQUEST.value(), 400, "잘못된 요청"),

    SERVER_ERROR(HttpStatus.INTERNAL_SERVER_ERROR.value(), 500, "서버에러"),

    NULL_POINT(HttpStatus.INTERNAL_SERVER_ERROR.value(), 512, "Null point")

    ;

    private final Integer httpStatusCode;
    private final Integer errorCode;
    private final String description;

}

File: PART3/ch03/ch03_springcloudapigateway/ch03_03/service/common/src/main/kotlin/org/delivery/common/error/ErrorCode.kt
package org.delivery.common.error

enum class ErrorCode(
    private val httpStatusCode: Int,
    private val errorCode: Int,
    private val description: String,
) : ErrorCodeIfs {

    OK(200 , 200 , "성공"),
    BAD_REQUEST(400, 400, "잘못된 요청"),

    SERVER_ERROR(500, 500, "서버에러"),

    NULL_POINT(500, 512, "Null point")

    ;

    override fun getHttpStatusCode(): Int {
        return this.httpStatusCode
    }

    override fun getErrorCode(): Int {
        return this.errorCode
    }

    override fun getDescription(): String {
        return this.description
    }
}